application/views.py

```diff
@@ -1,37 +1,43 @@
 from django.http import HttpResponseRedirect
 from django.shortcuts import render
-from .forms import HighSchoolApplicationForm
-from .models import HighSchoolApplication
 from django.utils import timezone
 from django.urls import reverse
 
+from .forms import HighSchoolApplicationForm
+from .models import HighSchoolApplication
+from high_school.models import Program
+
 
 def new_application(request, user_type):
-    if request.method == "POST":
-        # TODO user_id will be replaced by sessions
-        user_id = 1
-        form = HighSchoolApplicationForm(request.POST)
-        if form.is_valid():
-            f = form.save(commit=False)
-            f.user_id = user_id
-            f.application_number = (
-                str(user_id) + str(f.school.school_name) + str(f.program)
-            )
-            f.submitted_date = timezone.now()
-            try:
-                if "Submit" in request.POST.get("submit"):
+    try:
+        if request.method == "POST":
+            # TODO user_id will be replaced by sessions
```

Comment:
Now that we have sessions implemented, we might want to retrieve the user from the request and put additional checks for handling unauth and invalid access.

Reply:
Refer to the comment above